cmd/tools/grafana/dashboard_test.go

```diff
@@ -783,6 +780,10 @@ func checkTopKRange(t *testing.T, path string, data []byte) {
 				}
 			}
 		}
+
+		if strings.Contains(expr.expr, "$__range") && strings.Contains(expr.expr, "@ end()") {
```

Comment:
Since the query does not require whitespace, probably safer to go with this check to reduce false positives

noWhitespace := strings.ReplaceAll(expr.expr, " ", "")
if strings.Contains(noWhitespace, "$__range@end()") {
  hasRange = true
}